presto-main/src/main/java/com/facebook/presto/operator/scalar/MathFunctions.java

```diff
@@ -477,6 +528,24 @@ public static double round(@SqlType(StandardTypes.DOUBLE) double num, @SqlType(S
         return Math.round(num * factor) / factor;
     }
 
+    @Description("round to given number of decimal places")
+    @ScalarFunction("round")
+    @SqlType(StandardTypes.FLOAT)
+    public static long roundFloat(@SqlType(StandardTypes.FLOAT) long num, @SqlType(StandardTypes.BIGINT) long decimals)
+    {
+        float numInFloat = intBitsToFloat((int) num);
+        if (Float.isNaN(numInFloat) || Float.isInfinite(numInFloat)) {
+            return num;
+        }
+
+        double factor = Math.pow(10, decimals);
```

Comment:
Since we are using `double` factor here, shouldn't we maybe just call `round(double)` and cast return type to float?
`Double.Infinity` casts properly to `Float.Infinity`.